### PR TITLE
record editor config

### PR DESCRIPTION
**Pull request by zzacharo**

### CODE REVIEW

File: invenio_records_editor/config.py [hunk 1]
```diff
@@ -18,3 +18,63 @@
 
 RECORDS_EDITOR_TEMPLATE = "invenio_records_editor/editor.html"
 """Default editor template."""
+
+RECORDS_EDITOR_UI_CONFIG = {
+    "json-editor": {
```
Comment (topless) [inline invenio_records_editor/config.py]: `editor_config` for the shake of consistency.

File: invenio_records_editor/templates/invenio_records_editor/editor.html [hunk 2]
```diff
@@ -21,7 +21,8 @@
     <link href="//stackpath.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css"
       rel="stylesheet" integrity="sha384-wvfXpqpZZVQGK6TAh5PVlGOfQNHSoD2xbE+QkPxCAFlNEevoEH3Sl0sibVcOQVnN"
       crossorigin="anonymous">
-  </head>
+      <div id="invenio-editor-config" data-config='{{ ui_config | default("{}") | tojson }}'></div>
```
Comment (topless) [inline invenio_records_editor/templates/invenio_records_editor/editor.html]: Default is redundant.

**Review by topless — COMMENTED** [review]
There is a failure in travis also, other than that it looks good.

File: invenio_records_editor/views.py [hunk 3]
```diff
@@ -25,14 +26,43 @@ def create_editor_blueprint(app):
...
+
+        ui_config.update(endpoint_config)
+        ui_config["record_config"]["apiUrl"] = "{0}{1}".format(
+            endpoint_config["record_config"]["apiUrl"], recid
+        )
+
+        return render_template(
+            app.config["RECORDS_EDITOR_TEMPLATE"],
+            editor_url=app.config["RECORDS_EDITOR_URL_PREFIX"],
+            module_name="Invenio-Records-Editor",
```
Comment (ntarocco) [inline invenio_records_editor/views.py:57]: Is `module_name` the title of the page? In this case, I would rename the param...

File: invenio_records_editor/views.py [hunk 4]
```diff
@@ -25,14 +26,43 @@ def create_editor_blueprint(app):
...
+        endpoint_config = json_editor_config.get(rec_type)
+
+        ui_config.update(endpoint_config)
+        ui_config["record_config"]["apiUrl"] = "{0}{1}".format(
+            endpoint_config["record_config"]["apiUrl"], recid
+        )
+
+        return render_template(
+            app.config["RECORDS_EDITOR_TEMPLATE"],
+            editor_url=app.config["RECORDS_EDITOR_URL_PREFIX"],
```
Comment (ntarocco) [inline invenio_records_editor/views.py:56]: This `editor_url` resolved to `url prefix` looks weird, no?

File: invenio_records_editor/config.py [hunk 5]
```diff
@@ -18,3 +18,92 @@
...
+
+RECORDS_EDITOR_UI_CONFIG = {}
+"""Editor ui config.
+
+Example given:
+
+RECORDS_EDITOR_UI_CONFIG = {
+    "items": {
+        "record_config": {
+            "apiUrl": "/api/items/",
```
Comment (ntarocco) [inline invenio_records_editor/config.py]: do you have to split `list` and `item`?

**Review by ntarocco — APPROVED** [review]

File: invenio_records_editor/views.py [hunk 6]
```diff
@@ -25,14 +26,43 @@ def create_editor_blueprint(app):
...
     @need_editor_permissions("editor-view")
-    def index():
+    def edit_record(rec_type, recid):
         """Render a basic view, with dummy permission editor-view."""
+        ui_config = {}
+        json_editor_config = current_app.config["RECORDS_EDITOR_UI_CONFIG"]
+        endpoint_config = json_editor_config.get(rec_type)
+
+        ui_config.update(endpoint_config)
+        ui_config["record_config"]["apiUrl"] = "{0}{1}".format(
```
Comment (kpsherva) [inline invenio_records_editor/views.py]: my OCD related: any reason for camelCase and underscore in the same dict?

**Review by kpsherva — APPROVED** [review]

**Comment by ntarocco** [comment]
@zzacharo I think this PR is obsolete and to be closed, right?

**Comment by zzacharo** [comment]
Yes sorry for that!

**Pull request closed without merging**